mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -28,26 +28,32 @@
 class KiViBufferWalker():
     """! Simple auxiliary class used to walk through a buffer and search for KV tokens """
     def __init__(self):
-        self.KIVI_REGEX = r"\{\{([\w\d_-]+);([^\}]+)\}\}\n"
+        self.KIVI_REGEX = r"\{\{([\w\d_-]+);([^\}]+)\}\}"
         self.buff = str()
-        self.buff_idx = 0
+        self.kvl = []
         self.re_kv = re.compile(self.KIVI_REGEX)
 
     def append(self, payload):
-        """! Append stream buffer with payload """
+        """! Append stream buffer with payload and process"""
         self.buff += payload
+        lines = self.buff.split('\n')
```

Comment:
Since ```\n``` is removed from K,V regex any character outside the K.V pattern does not affect us. I have tested it. It works with mbed-os master and ARMmbed/mbed-os#2702